packages/app/src/AutocompleteInput.tsx

```diff
@@ -154,6 +180,39 @@ export default function AutocompleteInput({
               {belowSuggestions}
             </div>
           )}
+          <div>
+            {
+              // only show search history when: 1.no input, 2.has search type, 3.has history list
+              value.length === 0 &&
+                queryHistoryType &&
+                queryHistoryList.length > 0 && (
+                  <div className="border-top border-dark fs-8 py-2">
```

Comment:
nit: for readability and transparency, you could make this a var
`const shouldShowSearchHistory = value.length === 0 && queryHistoryType && queryHistoryList.length > 0;`